src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -1,20 +1,26 @@
-import PropTypes from "prop-types";
 import {observer} from "mobx-react-lite";
-import {useState} from "react";
-import ProgressDrawer from "../ProgressDrawer/ProgressDrawer";
+import React, {useState} from "react";
 import Badge from "../Badge/Badge";
+import ProgressDrawer from "../ProgressDrawer/ProgressDrawer";
+import {PlayList} from "../../models/play-list.model";
 
+interface ProgressBarProps {
+    highScore: number
+    playList: PlayList,
+    scenarioName: string,
+    scores: []
+}
 
-const ProgressBar = observer((props) => {
+const ProgressBar = (props: ProgressBarProps) => {
     const [isOpen, toggleOpen] = useState(false);
 
     const highScore = props.highScore;
-    const reqs = props.reqs.scenarioList.find(s => s.name === props.scenario).reqs;
+    const reqs = props.playList.scenarioList.find(s => s.name === props.scenarioName).reqs;
     const [nextRank, nextRankReq] = highScore > 0 ?
         Object.entries(reqs).find(req => req[1] > highScore) || ["max", highScore]
         : ["silver", reqs["silver"]];
     const [currentRank, currentRankReq] = Object.entries(reqs).filter(req => req[1] <= highScore).pop() || ["unranked", 0];
-    const percentToLevel = 100 * (highScore / nextRankReq) || 0
+    const percentToLevel = 100 * (highScore / +nextRankReq) || 0
```

Comment:
what does the + do here

Reply:
This is just short hand for number coercion. Essentially just making sure that value is a number.